ruwind/src/machine.rs

```diff
@@ -12,10 +13,12 @@ impl Machine {
         process: Process) -> bool {
         match self.processes.entry(pid) {
             Vacant(entry) => {
+                debug!("Process added: pid={}", pid);
                 entry.insert(process);
                 true
             },
             Occupied(_) => {
+                warn!("Process addition failed: pid={} already exists", pid);
```

Comment:
```suggestion
                debug!("Process already exists: pid={}", pid);
```

This is a common pattern where we receive data that might have duplicate entries, and so we deduplicate it here.